radarpipeline/project/project.py

```diff
@@ -300,36 +341,26 @@ def _get_total_required_data(self) -> List[str]:
         for feature_group in self.feature_groups:
             total_required_data.update(feature_group.get_required_data())
 
+        logger.info(f"Total required data: {total_required_data}")
+
         return list(total_required_data)
 
     def fetch_data(self) -> None:
         """
         Fetches the data from the data source
         """
 
-        if self.config["input_data"]["data_location"] == "sftp":
-            if self.config["input_data"]["data_format"] == "csv":
-                # self.data = SFTPDataReaderCSV(
-                #     self.config["input_data"], self.total_required_data
-                # ).read()
-                pass
-            else:
-                raise ValueError("Wrong data format")
-
-        elif self.config["input_data"]["data_location"] == "local":
+        if self.config["input_data"]["data_location"] == "local":
             if self.config["input_data"]["data_format"] == "csv":
                 self.data = SparkCSVDataReader(
                     self.config["input_data"], self.total_required_data
                 ).read()
```

Comment:
For some reason, `SparkCSVReader` is not reading `value.time `and `value.timeReceived`. I think it has something to do with Schema.

Reply:
Strange. I'll check it out and see why its happening.

Reply:
Hey @Hsankesara,

Seems like `SoarkCSVReader` is able to read the data with the time values as seen in the output below:
<img width="1021" alt="image" src="https://user-images.githubusercontent.com/42088801/189517138-83773c05-915e-4f89-b98b-1aaee523a861.png">

When this data is being sent to the features module, it is unable to read it as is is asking to use backticks to access the variable:
```
 Cannot resolve column name "value.time" among (key.projectId, key.userId, key.sourceId, value.time, value.timeReceived, value.batteryLevel, value.isPlugged, value.status); did you mean to quote the `value.time` column?
```

This would be solved if we can send the data as a pandas dataframe by default (as pandas is more widely used than spark DFs) for the feature processing, and convert it back to spark dataframe when it is processed  It is related to this earlier comment of yours:
> One more change we can do is ask the research what input format they are using, So we can convert pyspark df to suitable format. Mock-feature uses pandas, so we should convert it to pandas df. But more generic module would be helpful.

There are a few more considerations that we have to keep in mind:
1. We should also allow the user to use feature processing on the Spark DF if they wish to through specifying it in the config (as you said earlier). But in these cases, the user should be aware of the typicalities of Spark DFs, such as this bakctick issue (though the error messages are helpful in identifying them). Can we go forward with the assumptions that the user would be aware of these if they are explicitly specifying to use the Spark DF through the config?
2. Converting Spark DF to Pandas and back to Spark would cause some computation overhead, which is fine for small data but can take a significant amount of time for larger data. I think we should go forward with the conversion to Pandas for now and can later work on optimizations/other methods to decrease this overhead.

Reply:
Hey Pushkar,

When I was testing the module, I converted the spark data frame to pandas data frame, and it wasn't showing the time values for some reason. I will check it again. Maybe it's because of the backtick issue. 

I agree, we should allow user to choose what they want to use. If they are comfortable in using pySpark, then that should be encouraged. But yeah, let's go forward with converting it to pandas df, and we'll focus on optimazation later.

Reply:
Just tested it again, seems like the problem is still there. 
<img width="1111" alt="Screenshot 2022-09-12 at 11 02 05" src="https://user-images.githubusercontent.com/21009715/189627095-790f67ec-5029-4e39-b5d4-79559d6a839d.png">